File: codility/counting_elements/PermCheck.js
'use strict';

function solution(a) {
	a.sort((a, b) => a - b);

	var length = a.length;

	for(let i = 1; i < length; i++) {

		if(a[i] != a[i - 1] + 1 && a[i] !== a[i - 1]) {
			return 0;
		}
	}

	return 1;
}




let array = [4, 1, 1, 2, 3, 3, 9, 8, 6, 5];

console.log(solution(array));

File: codility/counting_elements/PermCheck.rb
def solution(a)
	a.sort!

	length = a.length

	(1..length).each do |i|
		if a[i] !== a[1 - 1] + 1 && a[i] !== a[i - 1]
			return 0
		end
	end

	return 1
end


array = [4, 1, 1, 2, 3, 3, 9, 8, 6, 5]

puts solution(array)